fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/rewrite/RewriteBottomUpJob.java

```diff
@@ -72,12 +79,14 @@ public void execute() throws AnalysisException {
             return;
         }
 
+        trace(logicalExpression);
         List<Rule> validRules = getValidRules(logicalExpression, rules);
         for (Rule rule : validRules) {
             GroupExpressionMatching groupExpressionMatching
                     = new GroupExpressionMatching(rule.getPattern(), logicalExpression);
             for (Plan before : groupExpressionMatching) {
-                Optional<CopyInResult> copyInResult = invokeRewriteRuleWithTrace(rule, before, group);
+                Optional<CopyInResult> copyInResult = invokeRewriteRuleWithTrace(rule, before, group,
+                        REWRITE_BOTTOM_UP_JOB_TRACER);
```

Comment:
you could use a mixin interface to get the tracer from a method from this interface and to avoid this parameter.
```java
RewriteBottomUpJob extends Job implements TracerSupplier {
    ...
}

interface TracerSupplier {
    EventProducer getTracer();
}
```

Reply:
nice!